src/Module.Server/Common/Models/CrpgAgentApplyDamageModel.cs

```diff
@@ -195,6 +195,49 @@ public override float GetDamageMultiplierForBodyPart(BoneBodyPartType bodyPart,
         defenderStunMultiplier = 1f;
     }
 
+    // This method should be adjuste once totalenergy uses our strike model to adjust the result when the strike is very bad.
+    public override bool DecideCrushedThrough(
+        Agent attackerAgent,
+        Agent defenderAgent,
+        float totalAttackEnergy,
+        Agent.UsageDirection attackDirection,
+        StrikeType strikeType,
+        WeaponComponentData defendItem,
+        bool isPassiveUsage)
+    {
+        EquipmentIndex wieldedItemIndex = attackerAgent.GetWieldedItemIndex(Agent.HandIndex.OffHand);
+        if (wieldedItemIndex == EquipmentIndex.None)
+        {
+            wieldedItemIndex = attackerAgent.GetWieldedItemIndex(Agent.HandIndex.MainHand);
+        }
+
+        var weaponComponentData = wieldedItemIndex != EquipmentIndex.None
+            ? attackerAgent.Equipment[wieldedItemIndex].CurrentUsageItem
+            : null;
+        if (weaponComponentData == null ||
+            isPassiveUsage ||
+            !weaponComponentData.WeaponFlags.HasAnyFlag(WeaponFlags.CanCrushThrough) ||
+            strikeType != StrikeType.Swing
+            || attackDirection != Agent.UsageDirection.AttackUp)
+        {
+            return false;
+        }
+
+        
+        int attackerPowerStrikeSkill = GetSkillValue(attackerAgent.Origin, CrpgSkills.PowerStrike);
+
+        int defenderStrengthSkill = GetSkillValue(defenderAgent.Origin, CrpgSkills.Strength);
+        int defenderShieldSkill = GetSkillValue(defenderAgent.Origin, CrpgSkills.Shield);
+        float defenderDefendPower = Math.Max(defenderShieldSkill * 6 + 3, defenderStrengthSkill);
+        int randomNumber = MBRandom.RandomInt(0, 1001);
+        if (defendItem != null && defendItem.IsShield)
+        {
+            return (randomNumber / 10f) > Math.Pow(attackerPowerStrikeSkill / defenderDefendPower, 2) * 100f;
+        }
+
+        return (randomNumber / 10f) > Math.Pow(attackerPowerStrikeSkill / defenderStrengthSkill, 2) * 100f;
```

Comment:
When the formula is repeated twice it can be understood as two very different formulas when in this case only one member change so by doing the following refactoring you reduce the code, you don't repeat yourself (#DRY) but most importantly you emphasize what differs when the defender has a shield.
```suggestion
        int defenderStrengthSkill = GetSkillValue(defenderAgent.Origin, CrpgSkills.Strength);
        int defenderShieldSkill = GetSkillValue(defenderAgent.Origin, CrpgSkills.Shield);
        float defenderDefendPower = defendItem != null && defendItem.IsShield
            ? Math.Max(defenderShieldSkill * 6 + 3, defenderStrengthSkill)
            : defenderStrengthSkill;
        int randomNumber = MBRandom.RandomInt(0, 1001);
        return randomNumber / 10f > Math.Pow(attackerPowerStrikeSkill / defenderDefendPower, 2) * 100f;
```